src/Compilers/CSharp/Test/Syntax/Parsing/WithElementParsingTests.cs

```diff
@@ -0,0 +1,2336 @@
+                        }
+                    }
+                }
+            }
+            N(SyntaxKind.CloseBracketToken);
+        }
+        EOF();
+    }
+
+    [Theory, MemberData(nameof(CollectionArgumentsLanguageVersions))]
+    public void NotWithElement9(LanguageVersion languageVersion)
+    {
+        UsingExpression("[with++]",
+            TestOptions.Regular.WithLanguageVersion(languageVersion));
+
+        N(SyntaxKind.CollectionExpression);
+        {
+            N(SyntaxKind.OpenBracketToken);
+            N(SyntaxKind.ExpressionElement);
+            {
+                N(SyntaxKind.PostIncrementExpression);
+                {
+                    N(SyntaxKind.IdentifierName);
+                    {
+                        N(SyntaxKind.IdentifierToken, "with");
+                    }
+                    N(SyntaxKind.PlusPlusToken);
+                }
+            }
+            N(SyntaxKind.CloseBracketToken);
+        }
+        EOF();
+    }
+
+    [Theory, MemberData(nameof(CollectionArgumentsLanguageVersions))]
+    public void NotWithElement10(LanguageVersion languageVersion)
+    {
+        UsingExpression("[with)]",
+            TestOptions.Regular.WithLanguageVersion(languageVersion),
+            // (1,6): error CS1003: Syntax error, ',' expected
+            // [with)]
+            Diagnostic(ErrorCode.ERR_SyntaxError, ")").WithArguments(",").WithLocation(1, 6));
+
+        N(SyntaxKind.CollectionExpression);
+        {
+            N(SyntaxKind.OpenBracketToken);
+            N(SyntaxKind.ExpressionElement);
+            {
+                N(SyntaxKind.IdentifierName);
+                {
+                    N(SyntaxKind.IdentifierToken, "with");
+                }
+            }
+            N(SyntaxKind.CloseBracketToken);
+        }
+        EOF();
+    }
+
+    [Theory, MemberData(nameof(CollectionArgumentsLanguageVersions))]
+    public void NotWithElement11(LanguageVersion languageVersion)
+    {
+        UsingExpression("[with..with]",
+            TestOptions.Regular.WithLanguageVersion(languageVersion));
+
+        N(SyntaxKind.CollectionExpression);
+        {
+            N(SyntaxKind.OpenBracketToken);
+            N(SyntaxKind.ExpressionElement);
+            {
+                N(SyntaxKind.RangeExpression);
+                {
+                    N(SyntaxKind.IdentifierName);
+                    {
+                        N(SyntaxKind.IdentifierToken, "with");
+                    }
+                    N(SyntaxKind.DotDotToken);
+                    N(SyntaxKind.IdentifierName);
+                    {
+                        N(SyntaxKind.IdentifierToken, "with");
+                    }
+                }
+            }
+            N(SyntaxKind.CloseBracketToken);
+        }
+        EOF();
+    }
+
+    [Theory, MemberData(nameof(CollectionArgumentsLanguageVersions))]
+    public void NotWithElement12(LanguageVersion languageVersion)
+    {
+        UsingExpression("[with..with()]",
```

Comment:
Perhaps `[a.with()]` is also worth testing as well as `[(with)()]`

Reply:
Sure.  Can i add in https://github.com/dotnet/roslyn/pull/80495?

Reply:
Yes

Reply:
62aceeb2208